agent/boundarylogproxy/proxy_test.go

```diff
@@ -0,0 +1,588 @@
+//go:build linux || darwin
+
+package boundarylogproxy_test
+
+import (
+	"context"
+	"encoding/binary"
+	"net"
+	"path/filepath"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/coder/coder/v2/agent/boundarylogproxy"
+	agentproto "github.com/coder/coder/v2/agent/proto"
+	"github.com/coder/coder/v2/testutil"
+)
+
+// sendMessage writes a length-prefixed protobuf message to the connection.
+func sendMessage(t *testing.T, conn net.Conn, req *agentproto.ReportBoundaryLogsRequest) {
+	t.Helper()
+
+	data, err := proto.Marshal(req)
+	if err != nil {
+		//nolint:gocritic // In tests we're not worried about conn being nil.
+		t.Errorf("%s marshal req: %s", conn.LocalAddr().String(), err)
+	}
+
+	var header uint32
+	//nolint:gosec // In tests we're not worried about possible integer overflow.
+	header |= uint32(len(data))
+	header |= 1 << 28
+
+	err = binary.Write(conn, binary.BigEndian, header)
+	if err != nil {
+		//nolint:gocritic // In tests we're not worried about conn being nil.
+		t.Errorf("%s write conn length: %s", conn.LocalAddr().String(), err)
+	}
+
+	_, err = conn.Write(data)
+	if err != nil {
+		//nolint:gocritic // In tests we're not worried about conn being nil.
+		t.Errorf("%s write conn data: %s", conn.LocalAddr().String(), err)
+	}
```

Comment:
Duplicated code from `sendBoundaryLogsRequest`; see my [review](https://github.com/coder/boundary/pull/124#pullrequestreview-3610164318) of the boundary code which suggests extract the framing into its own package for reuse.

Reply:
Agreed, I will commonize this.